pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -786,6 +786,7 @@ type NodeStatus struct {
 	WorkflowTemplateName string `json:"workflowTemplateName,omitempty" protobuf:"bytes,19,opt,name=workflowTemplateName"`
 
 	// TemplateScope is the template scope in which the template of this node was retrieved.
+	// DEPRECATED: This value is not used anymore.
```

Comment:
Note that I now deprecate `TemplateScope`. This is unrelated to this change, but I noticed it was not used anymore. Please comment if this is not the case.

Reply:
It is a reference of the context base of template resolution. If a node doesn't have this information, it's not possible to retrieve `template`, not `templateRef`, unless we parse everything from the tree root in the future. What do you think?

Reply:
Reverted